packages/astro/src/core/build/index.ts

```diff
@@ -33,6 +33,8 @@ import { collectPagesData } from './page-data.js';
 import { staticBuild, viteBuild } from './static-build.js';
 import type { StaticBuildOptions } from './types.js';
 import { getTimeStat } from './util.js';
+import { basename } from 'node:path';
```

Comment:
If it is at all possible to avoid working with paths, I would highly recommend sticking with URLs for the implementation. Windows always bites us.